shopgate/pwa
# Migrate ScrollHeader changes from PWA6

# Description

During the development of PWA7 the `ScrollHeader` component was refactored in PWA6, but the changes where not transferred to PWA7 at that time. This pull request makes up for this.

The ScrollHeader is used to toggle visibility of the `FilterBar` when users scroll inside a product list. In PWA6 the system how to listen for scroll events was refactored, so that it could be also used for the `TabBar` in the ios 11 theme (this was already migrated earlier). Unlike in earlier PWA versions, it's now also possible to deactivate hiding of the FilterBar via a "widget config".

## Type of change

## How to test it

Navigate to a product list with filters and scroll up / down. The filter bar should still hide and show depending on the scrolling. It should be possible to deactivate scrolling via the `@shopgate/engage/components/FilterBar` widget setting.
